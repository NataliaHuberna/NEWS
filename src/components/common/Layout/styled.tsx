import styled from "styled-components";
import {COLORS} from '../styles/colors';

export const StLayout = styled.div`
  min-height: 100vh;
  & > div{
    position: relative;
  }
  &>div:nth-child(1){
    height: 80px;
    box-shadow: 1px 1px 7px 0 ${COLORS.shadowBox};
    z-index: 1;
  }
  &>div:nth-child(2){
    height: calc(100vh - 160px);
    overflow-y: auto;
  }
  &>div:nth-child(3){
    height: 80px;
  }
`;

export const StContent = styled.div`
  background-image: ${COLORS.purpleBg};
`;


