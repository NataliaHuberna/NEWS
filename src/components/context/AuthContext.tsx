import React, { useState } from 'react';

type TProps = {
    children: React.ReactNode
};

export const AuthContext = React.createContext(null);

const AuthProvider: React.FC<TProps> = ({ children}) => {
    const [user, setUser] = useState(null);
    const value = {
        user,
        setUser
    };
    
    return <AuthContext.Provider value={value}>
        {children}
    </AuthContext.Provider>;
};

export default AuthProvider;
