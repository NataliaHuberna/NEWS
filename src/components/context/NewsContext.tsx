import React, { useState } from 'react';
import {TNew} from "src/types/newsTypes";

type TProps = {
    children: React.ReactNode
};

export const NewsContext = React.createContext(null);

const NewsProvider: React.FC<TProps> = ({ children}) => {
    const [state, setState] = useState({news:[], singleNew: null});
    const setNews = (news: Array<TNew>) => setState((prevState) => ({...prevState, news}));
    const setSingleNew = (publishedAt: string) => setState((prevState) => ({...prevState,
        singleNew: prevState.news.find((singleNew: TNew) => singleNew.publishedAt === publishedAt )}));
    const value = {
        ...state,
        setNews,
        setSingleNew
    };
    return <NewsContext.Provider value={value}>
        {children}
    </NewsContext.Provider>;

};

export default NewsProvider;
