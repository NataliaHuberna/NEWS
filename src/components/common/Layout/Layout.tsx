import React from 'react';
import {Outlet} from "react-router";
import Footer from '../Footer/Footer';
import Header from '../Header/Header';
import {StContent, StLayout} from "./styled";

const Layout: React.FC = () => (
    <StLayout>
        <Header/>
        <StContent>
            <Outlet/>
        </StContent>
        <Footer/>
    </StLayout>
);

export default Layout;
