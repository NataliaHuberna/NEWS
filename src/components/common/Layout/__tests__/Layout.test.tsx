import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter} from 'react-router-dom';
import { AuthContext } from '../../../context/AuthContext';
import Layout from '../Layout';

describe('Layout', () => {
    const mockAuthContextValue = {
        user: 'Some user',
        setUser: jest.fn()
    };

    const sut = (mockValue: any) => (
        mount(
            <BrowserRouter>
                <AuthContext.Provider value={mockValue}>
                    <Layout />
                </AuthContext.Provider>
            </BrowserRouter>
        ));
    it('should render Header Outlet and Footer', () => {
        const wrapper = sut(mockAuthContextValue);

        expect(wrapper.find('Header').exists()).toBe(true);
        expect(wrapper.find('Outlet').exists()).toBe(true);
        expect(wrapper.find('Footer').exists()).toBe(true);
    });

    it('should match....', () => {
        const wrapper = sut(mockAuthContextValue);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
