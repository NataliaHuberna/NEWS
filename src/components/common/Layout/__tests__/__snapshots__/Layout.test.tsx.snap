// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match.... 1`] = `
"<BrowserRouter>
  <Router basename={[undefined]} location={{...}} navigationType=\\"POP\\" navigator={{...}}>
    <Layout>
      <styled.div>
        <div className=\\"sc-gKclnd iHbEwV\\">
          <Header>
            <styled.div>
              <div className=\\"sc-jRQBWg jPddjv\\">
                <NavBar>
                  <styled.nav>
                    <nav className=\\"sc-hKwDye fwkzUq\\">
                      <ForwardRef(ButtonGroup) variant=\\"contained\\" aria-label=\\"outlined primary button group\\">
                        <MuiButtonGroupRoot as=\\"div\\" role=\\"group\\" className=\\"MuiButtonGroup-root MuiButtonGroup-contained\\" ownerState={{...}} aria-label=\\"outlined primary button group\\">
                          <Noop />
                          <div role=\\"group\\" className=\\"MuiButtonGroup-root MuiButtonGroup-contained css-zqcytf-MuiButtonGroup-root\\" aria-label=\\"outlined primary button group\\">
                            <ForwardRef(Button) onClick={[Function: onClick]} variant=\\"contained\\" disabled={true}>
                              <MuiButtonRoot ownerState={{...}} className=\\"MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary\\" component=\\"button\\" disabled={true} focusRipple={true} focusVisibleClassName=\\"\\" type={[undefined]} onClick={[Function: onClick]} disableRipple={false} classes={{...}}>
                                <Noop />
                                <ForwardRef(ButtonBase) className=\\"MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-1yu95zv-MuiButton-root\\" component=\\"button\\" disabled={true} focusRipple={true} focusVisibleClassName=\\"\\" type={[undefined]} onClick={[Function: onClick]} disableRipple={false} classes={{...}}>
                                  <MuiButtonBaseRoot as=\\"button\\" className=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-1yu95zv-MuiButton-root\\" ownerState={{...}} onBlur={[Function (anonymous)]} onClick={[Function: onClick]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={-1} type=\\"button\\" disabled={true} classes={{...}}>
                                    <Noop />
                                    <button className=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-sghohy-MuiButtonBase-root-MuiButton-root\\" onBlur={[Function (anonymous)]} onClick={[Function: onClick]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={-1} type=\\"button\\" disabled={true}>
                                      Home
                                    </button>
                                  </MuiButtonBaseRoot>
                                </ForwardRef(ButtonBase)>
                              </MuiButtonRoot>
                            </ForwardRef(Button)>
                            <ForwardRef(Button) onClick={[Function: onClick]} variant=\\"contained\\" disabled={false}>
                              <MuiButtonRoot ownerState={{...}} className=\\"MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary\\" component=\\"button\\" disabled={false} focusRipple={true} focusVisibleClassName=\\"\\" type={[undefined]} onClick={[Function: onClick]} disableRipple={false} classes={{...}}>
                                <Noop />
                                <ForwardRef(ButtonBase) className=\\"MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-1yu95zv-MuiButton-root\\" component=\\"button\\" disabled={false} focusRipple={true} focusVisibleClassName=\\"\\" type={[undefined]} onClick={[Function: onClick]} disableRipple={false} classes={{...}}>
                                  <MuiButtonBaseRoot as=\\"button\\" className=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-1yu95zv-MuiButton-root\\" ownerState={{...}} onBlur={[Function (anonymous)]} onClick={[Function: onClick]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={0} type=\\"button\\" disabled={false} classes={{...}}>
                                    <Noop />
                                    <button className=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-sghohy-MuiButtonBase-root-MuiButton-root\\" onBlur={[Function (anonymous)]} onClick={[Function: onClick]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={0} type=\\"button\\" disabled={false}>
                                      All News
                                      <ForwardRef(TouchRipple) center={false}>
                                        <MuiTouchRippleRoot className=\\"MuiTouchRipple-root\\">
                                          <Noop />
                                          <span className=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\">
                                            <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                          </span>
                                        </MuiTouchRippleRoot>
                                      </ForwardRef(TouchRipple)>
                                    </button>
                                  </MuiButtonBaseRoot>
                                </ForwardRef(ButtonBase)>
                              </MuiButtonRoot>
                            </ForwardRef(Button)>
                            <ForwardRef(Button) id=\\"logoutButton\\" variant=\\"contained\\" onClick={[Function: handleClickLogout]}>
                              <MuiButtonRoot ownerState={{...}} className=\\"MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary\\" component=\\"button\\" disabled={false} focusRipple={true} focusVisibleClassName=\\"\\" type={[undefined]} id=\\"logoutButton\\" onClick={[Function: handleClickLogout]} disableRipple={false} classes={{...}}>
                                <Noop />
                                <ForwardRef(ButtonBase) className=\\"MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-1yu95zv-MuiButton-root\\" component=\\"button\\" disabled={false} focusRipple={true} focusVisibleClassName=\\"\\" type={[undefined]} id=\\"logoutButton\\" onClick={[Function: handleClickLogout]} disableRipple={false} classes={{...}}>
                                  <MuiButtonBaseRoot as=\\"button\\" className=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-1yu95zv-MuiButton-root\\" ownerState={{...}} onBlur={[Function (anonymous)]} onClick={[Function: handleClickLogout]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={0} type=\\"button\\" disabled={false} id=\\"logoutButton\\" classes={{...}}>
                                    <Noop />
                                    <button className=\\"MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary css-sghohy-MuiButtonBase-root-MuiButton-root\\" onBlur={[Function (anonymous)]} onClick={[Function: handleClickLogout]} onContextMenu={[Function (anonymous)]} onFocus={[Function (anonymous)]} onKeyDown={[Function (anonymous)]} onKeyUp={[Function (anonymous)]} onMouseDown={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} onMouseUp={[Function (anonymous)]} onDragLeave={[Function (anonymous)]} onTouchEnd={[Function (anonymous)]} onTouchMove={[Function (anonymous)]} onTouchStart={[Function (anonymous)]} tabIndex={0} type=\\"button\\" disabled={false} id=\\"logoutButton\\">
                                      Log out
                                      <ForwardRef(TouchRipple) center={false}>
                                        <MuiTouchRippleRoot className=\\"MuiTouchRipple-root\\">
                                          <Noop />
                                          <span className=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\">
                                            <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                          </span>
                                        </MuiTouchRippleRoot>
                                      </ForwardRef(TouchRipple)>
                                    </button>
                                  </MuiButtonBaseRoot>
                                </ForwardRef(ButtonBase)>
                              </MuiButtonRoot>
                            </ForwardRef(Button)>
                          </div>
                        </MuiButtonGroupRoot>
                      </ForwardRef(ButtonGroup)>
                    </nav>
                  </styled.nav>
                </NavBar>
              </div>
            </styled.div>
          </Header>
          <styled.div>
            <div className=\\"sc-iCfMLu lkSrtN\\">
              <Outlet />
            </div>
          </styled.div>
          <Footer>
            <styled.div>
              <div className=\\"sc-bdvvtL cjbcZU\\">
                <styled.div>
                  <div className=\\"sc-gsDKAQ gpLSln\\">
                    <styled.strong>
                      <strong className=\\"sc-dkPtRN eHRfMh\\">
                        All Rights Reserved, 
                        2022
                      </strong>
                    </styled.strong>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </Footer>
        </div>
      </styled.div>
    </Layout>
  </Router>
</BrowserRouter>"
`;

exports[`Layout test should render correctly 1`] = `ShallowWrapper {}`;
